Lvmdbusd test fixes

tasleson opened the pull request:
Changes needing to allow the tests to run in the lvm test environment.

Forced update to pickup circle CI for testing and a couple more fixes.

tasleson committed:
circleci: Add test script to run lvmdbusd tests

tasleson committed:
circleci: Add config.yaml

tasleson committed:
lvmdbustest.py: Correct use of lvmconfig

We need to leverage our LVM_EXECUTABLE variable so that we use what is
specified.

Signed-off-by: Tony Asleson <tasleson@redhat.com>

tasleson committed:
lvmdbustest.py: Handle changes needed for lvm test env.

Lvm testing can at times use a single loop back device which is segemented
into N number of linear segments.  This results in PVs like:

  PV                           VG Fmt  Attr PSize PFree
  /dev/mapper/LVMTEST481999pv1    lvm2 ---  6.25g 6.25g
  /dev/mapper/LVMTEST481999pv2    lvm2 ---  6.25g 6.25g
  /dev/mapper/LVMTEST481999pv3    lvm2 ---  6.25g 6.25g
  /dev/mapper/LVMTEST481999pv4    lvm2 ---  6.25g 6.25g
  /dev/mapper/LVMTEST481999pv5    lvm2 ---  6.25g 6.25g
  /dev/mapper/LVMTEST481999pv6    lvm2 ---  6.25g 6.25g

This broke some of the expected logic in the test used for nested testing.

Signed-off-by: Tony Asleson <tasleson@redhat.com>

tasleson committed:
udevwatch.py: Handle udev event['DEVNAME'] missmatch

Lvm testing can at times use a single loop back device which is segemented
into N number of linear segments.  This results in PVs which have a returned
path from lvm like: /dev/mapper/LVMTEST481999pv1.  When we get the udev
change event the event['DEVNAME'] is /dev/dm-N.  Thus we failed to find the
device in our in memory data struture and did not refresh our state.  In
the case where we fail to match on event['DEVNAME'] we will leverage
event['DEVLINKS'] which are all symbolic links pointing to the same device
node.

Signed-off-by: Tony Asleson <tasleson@redhat.com>

tasleson commented:
@vojtechtrefny Please review.  The last 3 commits won't be part of any commit series, so they can be skipped.

tasleson commented:
Hmm, I thought I had the circleCI stuff working before...

tasleson committed:
circleci: Make lvmdbusd executable

Signed-off-by: Tony Asleson <tasleson@redhat.com>

vojtechtrefny reviewed (approved):
Looks good to me.

tasleson committed:
udevwatch.py: Handle block devices outside of /dev

Signed-off-by: Tony Asleson <tasleson@redhat.com>

tasleson committed:
lvmdbustest.py: Increase number of LV

In testing 128 never worked.

Signed-off-by: Tony Asleson <tasleson@redhat.com>

tasleson committed:
lvmdbustest.py: Rename tests which need udev support

Allows developer to run the tests together using "-k udev" on the
command line, e.g.

$ ./lvmdbustest.py -v -k udev

Signed-off-by: Tony Asleson <tasleson@redhat.com>

tasleson committed:
lvmdbustest.py: Add a bit more context to asserts

Helps to understand if we were expecting or not expecting something to
be present.

Signed-off-by: Tony Asleson <tasleson@redhat.com>

tasleson committed:
lvmdbustest.py: Ensure writes to block devices occur in /dev

In some of the lvm testing they lvm devices are not in /dev.  We need to
locate and use the device mapper device with the matching major & minor to
use so that udev events are generated for the IO as udev only monitors
/dev.

Signed-off-by: Tony Asleson <tasleson@redhat.com>

tasleson commented:
@vojtechtrefny I added some additional commits, please check these newly added ones too.  Thanks!

vojtechtrefny commented on test/dbus/lvmdbustest.py:
```diff
@@ -2437,11 +2437,14 @@ def _block_present_absent(self, block_device, present=False):
 		if present:
 			rc = self._lookup(block_device)
-			self.assertNotEqual(rc, '/', "Daemon failed to update, missing udev change event?")
+			self.assertNotEqual(rc, '/', f"Daemon failed to update, block device{block_device} expecting to be"
```
Just a nitpick: missing space in `block device{block_device}`

vojtechtrefny reviewed (approved):
LGTM

tasleson committed:
lvmdbusd: Exit faster

Add a global timeout value to be used for the threads to end waiting for
whatever it is they are blocked on.  The values varied from 2-5 seconds,
which is way longer than needed.  Value of 0.5 shows no CPU load when
service is running and is idle.

tasleson committed:
lvmdbustest.py: Skip if g_tmo != 0

This test doesn't have any additional value when running when g_tmo != 0.

tasleson committed:
lvmdbustest.py: test_z_sigint: reduce #lv and intervals

We are trying to ensure that the daemon stops while it's busy processing
its internal queues.  Decrease the number of LVs and intervals to allow
this test to complete in less time.

tasleson committed:
lvmdbustest.py: Only check vdo support once

We don't need to repeat this in every setUp.
